lib/std/json/dynamic.zig

```diff
@@ -32,17 +32,25 @@ pub const Value = union(enum) {
     array: Array,
     object: ObjectMap,
 
-    pub fn parseFromNumberSlice(s: []const u8) Value {
+    pub fn parseFromNumberSlice(s: []const u8, options: ParseOptions) Value {
         if (!isNumberFormattedLikeAnInteger(s)) {
             const f = std.fmt.parseFloat(f64, s) catch unreachable;
             if (std.math.isFinite(f)) {
-                return Value{ .float = f };
+                if (options.parse_floats) {
+                    return Value{ .float = f };
+                } else {
+                    return Value{ .number_string = s };
+                }
             } else {
                 return Value{ .number_string = s };
             }
         }
         if (std.fmt.parseInt(i64, s, 10)) |i| {
-            return Value{ .integer = i };
+            if (options.parse_integers) {
+                return Value{ .integer = i };
+            } else {
+                return Value{ .number_string = s };
+            }
         } else |e| {
             switch (e) {
                 error.Overflow => return Value{ .number_string = s },
```

Comment:
When options say no parsing, seems like we should skip calling the parse() methods and immediately return a number string.  I'd imagine something like this:
```zig
    pub fn parseFromNumberSlice(s: []const u8, options: ParseOptions) Value {
        if (!isNumberFormattedLikeAnInteger(s)) {
            if (!options.parse_floats)
                return Value{ .number_string = s };
            const f = std.fmt.parseFloat(f64, s) catch unreachable;
            if (std.math.isFinite(f)) {
                return Value{ .float = f };
            } else {
                return Value{ .number_string = s };
            }
        }
        if (!options.parse_integers)
            return Value{ .number_string = s };
        if (std.fmt.parseInt(i64, s, 10)) |i| {
            return Value{ .integer = i };
        } else |e| {
            switch (e) {
                error.Overflow => return Value{ .number_string = s },
                error.InvalidCharacter => unreachable,
            }
        }
    }
```

Reply:
Better than what I wrote down

Going off of this, perhaps it would be even more optimized, and consistent with the current API, to implement two variations of json.Value rather than hijacking the json.ParseOptions struct?  I believe the intent behind the json.ParseOptions struct is that it's supposed to be independent of the type that you're parsing.

To my knowledge, it wouldn't interfere with existing code if the current json.Value struct was wrapped in a type-returning function, assuming an interoperable instance of that type is assigned to json.Value, right?

Doing the above would also have the benefit of allowing json.Value.parseFromNumberSlice to retain it's previous function signature.

Reply:
From what I can figure out, wrapping the json.Value enum in a type-function would require slight changes to the current JSON API, I'll wait and see what the maintainers say about all of this. It can be done, but it's not as simple as I'd hoped.